src/net.cpp

```diff
@@ -191,37 +191,40 @@ static int GetnScore(const CService& addr)
 }
 
 // Is our peer's addrLocal potentially useful as an external IP source?
-bool IsPeerAddrLocalGood(CNode *pnode)
+bool IsPeerAddrLocalGood(CNode& pnode)
 {
-    CService addrLocal = pnode->GetAddrLocal();
-    return fDiscover && pnode->addr.IsRoutable() && addrLocal.IsRoutable() &&
+    CService addrLocal = pnode.GetAddrLocal();
+    return fDiscover && pnode.addr.IsRoutable() && addrLocal.IsRoutable() &&
            IsReachable(addrLocal.GetNetwork());
 }
 
-// pushes our own address to a peer
-void AdvertiseLocal(CNode *pnode)
+void AdvertiseLocal(CNode& pnode)
 {
-    if (fListen && pnode->fSuccessfullyConnected)
-    {
-        CAddress addrLocal = GetLocalAddress(&pnode->addr, pnode->GetLocalServices());
-        if (gArgs.GetBoolArg("-addrmantest", false)) {
-            // use IPv4 loopback during addrmantest
-            addrLocal = CAddress(CService(LookupNumeric("127.0.0.1", GetListenPort())), pnode->GetLocalServices());
-        }
-        // If discovery is enabled, sometimes give our peer the address it
-        // tells us that it sees us as in case it has a better idea of our
-        // address than we do.
-        FastRandomContext rng;
-        if (IsPeerAddrLocalGood(pnode) && (!addrLocal.IsRoutable() ||
-             rng.randbits((GetnScore(addrLocal) > LOCAL_MANUAL) ? 3 : 1) == 0))
-        {
-            addrLocal.SetIP(pnode->GetAddrLocal());
-        }
-        if (addrLocal.IsRoutable() || gArgs.GetBoolArg("-addrmantest", false))
-        {
-            LogPrint(BCLog::NET, "AdvertiseLocal: advertising address %s\n", addrLocal.ToString());
-            pnode->PushAddress(addrLocal, rng);
-        }
+    assert(fListen);
+
+    // First consider our "best" addr for the peer, as seen locally.
+    // In absence of other information, we assume that the locally seen address
+    // is the "best" addr for peers to refer to our node.
+    CAddress addrLocal = GetLocalAddress(pnode.addr, pnode.GetLocalServices());
+
+    if (gArgs.GetBoolArg("-addrmantest", false)) {
+        // use IPv4 loopback during addrmantest
+        addrLocal = CAddress(CService(LookupNumeric("127.0.0.1", GetListenPort())), pnode.GetLocalServices());
+    }
+
+    // If discovery is enabled, we request that peers advertise our node to their peers. 
+    // Under some circumstances, our peers may have a more accurate perspective 
+    // on our address than our own node. Sometimes, we ask our peer to advertise 
+    // our node under the address they see us as at instead of the address that our 
```

Comment:
6dc9c7535daa931d7f747958d0dc6fa29417dd80 please remove trailing space to make linter happy:
```suggestion
    // If discovery is enabled, we request that peers advertise our node to their peers.
    // Under some circumstances, our peers may have a more accurate perspective
    // on our address than our own node. Sometimes, we ask our peer to advertise
    // our node under the address they see us as at instead of the address that our
```